### PR TITLE
Fix ert crashes if invalid range string in REPORT_STEPS config

### LINKED ISSUE #9057: Misconfigured configuration leads to ert crashing unexpectedly
### What happened? (You can include a screenshot if it helps explain)

Misconfigured GEN_DATA by adding: `REPORT_STEPS:H` (instead 0f `REPORT_STEPS:0`)

### What did you expect to happen?

Expected this to show in the Suggestor, instead I got: 
```
ERT crashed unexpectedly with "invalid literal for int() with base 10: 'H'".
See logfile(s) for details:
```

### steps to reproduce

Add:
```bash
GEN_DATA POLY_RES RESULT_FILE:poly.out REPORT_STEPS:H
```
to poly.ert

### Environment where bug has been observed

- [ ] python 3.8
- [ ] python 3.9
- [ ] python 3.10
- [ ] python 3.11
- [ ] python 3.12
- [ ] macosx
- [ ] rhel7
- [ ] rhel8
- [ ] local queue
- [ ] lsf queue
- [ ] slurm queue
- [ ] openPBS queue

### PR DESCRIPTION
**Pull request by HakonSohoel**
**Issue**
Resolves #9057

**Approach**
Added new test catching the bug and implemented a fix

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/config/gen_data_config.py [hunk 1]
```diff
@@ -55,16 +55,24 @@ def from_config_dict(cls, config_dict: ConfigDict) -> Optional[Self]:
...
                     f"Missing or unsupported RESULT_FILE for GEN_DATA key {name!r}",
                     name,
                 )
+            try:
+                _report_steps: Optional[List[int]] = rangestring_to_list(
+                    report_steps_value
+                )
+            except ValueError as e:
+                raise ConfigValidationError.with_context(
+                    f"The REPORT_STEPS setting: {report_steps_value} is invalid"
```
Comment (berland) [inline src/ert/config/gen_data_config.py:71]: Is it possible to also mention "POLY_RES" in this error message? Can make it easier to navigate to the right spot in the config file if there are many of these.

**Review by andreas-el — APPROVED** [review]
Very nice work! 🥓 
Just need to add release label & fix the PR title then we are ready to go! ✈️

File: src/ert/config/gen_data_config.py (hunk 1 above)
Comment (HakonSohoel) [reply src/ert/config/gen_data_config.py:71]: I can add that, but this is how the validation error looks like in the gui, so it seems like the user will get a good pointer to the right spot in the config file already:
image

Comment (HakonSohoel) [reply src/ert/config/gen_data_config.py:71]: Here is an alternative where "POLY_RES" is mentioned in the error message: 
image

Comment (berland) [reply src/ert/config/gen_data_config.py]: Yes, the latter looks very good!

**Comment by berland** [comment]
Remember to squash, we should not have the failing test commit as a separate commit 👍🏻

**Pull request merged**